lib/commands/log.js

```diff
@@ -98,23 +98,28 @@ export default {
         sim: this.device,
         udid: this.isRealDevice() ? this.opts.udid : undefined,
       });
-
-      if (this.isRealDevice()) {
-        this.logs.syslog = new IOSDeviceLog({
+      this.logs.syslog = this.isRealDevice()
+        ? new IOSDeviceLog({
           udid: this.opts.udid,
           showLogs: this.opts.showIOSLog,
           log: this.log,
-        });
-      } else {
-        this.logs.syslog = new IOSSimulatorLog({
+        })
+        : new IOSSimulatorLog({
           sim: /** @type {import('appium-ios-simulator').Simulator} */ (this.device),
           showLogs: this.opts.showIOSLog,
           iosSimulatorLogsPredicate: this.opts.iosSimulatorLogsPredicate,
           log: this.log,
         });
+      if (this.opts.showSafariConsoleLog) {
```

Comment:
Actually `showSafariConsoleLog`/`showSafariNetworkLog` should control if they show in Appium log, but they should be available via general get console logs/network logs always like syslog's `this.showLogs`

Reply:
I think it is not a good idea to always have them enabled by default, since log buffer need lots of memory and also CPU load is increased even if corresponding log buckets are not used. I could also update the error message while fetching them to advice enabling the flag

Reply:
Then, can we add one more `enableSafariConsoleLog` in addition to `showSafariConsoleLog`?
Showing Safari console log in appium.log is noisy in case we want to get safari console log via the log endpoint, but it could be helpful for some cases like `showiOSLog`. Or just `enableSafariConsoleLog` to keep current behavior, this looks reasonable to me.

Yea, then modifying the error message would be necessary

Reply:
improved the error message.
I don't really want to introduce a new variable, so the logic has been changed to disable the corresponding logs collection if the capability is unset. Otherwise if the capability is set to any boolean value then logs collection is enabled, it just decides whether to reflect them in the server log.

Reply:
Agree to not introduce new caps.
Then, what about re-using `showiOSLog`? So for example using both `enableSafariConsoleLog` plus `showiOSLog` prints them in appium server, but `enableSafariConsoleLog` only is for log endpoint use only.

Safari console log will be truncated (full console log could be a long line so truncating it itself is reasonable for me), so safari console log on appium server log does not help for log investigation. Users need to get the full log via the log endpoint anyway. Then, such appium server log's console log is just a noise, although it might help to sync timeline when safari console log line occurs.

Reply:
> Agree to not introduce new caps.
> Then, what about re-using showiOSLog? So for example using both enableSafariConsoleLog plus showiOSLog prints them in appium server, but enableSafariConsoleLog only is for log endpoint use only.


Sorry, I am confused. Wouldn't `enableSafariConsoleLog` be a new capability?

Reply:
Oh, sorry, `showSafariConsoleLog`. I mixed up with `enablePerformanceLogging` naming.

Reply:
Could you please describe in more details how the expected condition should work?
Is the current implementation not enough?

Reply:
Like https://github.com/appium/appium-xcuitest-driver/pull/2421/files#r1663012462, so I'd like to avoid showing safari console logs in appium server log to reduce noise but want to get such an application specific logs via log endpoint to handle errors etc in the console log for web app.